src/d_iwad.c

```diff
@@ -345,11 +348,97 @@ static void CheckInstallRootPaths(void)
 
 // Check for Doom downloaded via Steam
 
+#define MAX_STEAM_LIBRARIES 16
+
+static char *steam_libraries[MAX_STEAM_LIBRARIES];
+static int num_steam_libraries = 0;
+
+static void AddSteamLibraryDir(char *dir)
+{
+    if (num_steam_libraries < MAX_STEAM_LIBRARIES)
+    {
+        steam_libraries[num_steam_libraries] = M_StringReplace(dir, "\\\\",
+                                                               "\\");
+        ++num_steam_libraries;
+    }
+}
+
+static void CheckSteamLibrary(long library_file_size, char *library_file_buffer)
+{
+    long i, start;
```

Comment:
Please don't use type `long`, since it means very different things in different compilers. Please remove all uses of `long`.